app/policies/wiky_policy.rb

```diff
@@ -1,2 +1,53 @@
 class WikyPolicy < ApplicationPolicy
+    attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  def index?
+    user.present?
+  end
+
+  def show?
+    update?
+  end
+
+  def create?
+    user.present?
+  end
+
+  def new?
+    create?
+  end
+
+  def update?
+    user.present? && (record.public || user.admin? || record.user == user || record.collaborators.pluck(:user_id).include?(user.id))
```

Comment:
`pluck` usage can be removed and the final test simplified so this line becomes:
```ruby
user.present? && (record.public || user.admin? || record.user == user || record.collaborators.include?(user))
```